projects/storefrontlib/src/lib/product/components/product-list/product-facet-navigation/product-facet-navigation.component.spec.ts

```diff
@@ -32,4 +200,79 @@ describe('ProductFacetNavigationComponent in product-list', () => {
     component.toggleValue('mockQuery');
     expect(component.filter.emit).toHaveBeenCalledWith('mockQuery');
   });
+
+  describe('ProductFacetNavigationComponent UI tests', () => {
+    beforeEach(() => {
+      component.activeFacetValueCode = '0';
+      component.searchResult = {};
+      element = fixture.debugElement;
+    });
+
+    it(`should not show facet groups if there are no facets`, () => {
+      component.searchResult.facets = [];
+      fixture.detectChanges();
+
+      const facetGroups = element.queryAll(By.css('.y-search-facet-group'));
+      expect(facetGroups.length).not.toBeGreaterThan(0);
+    });
+
+    it(`should not show facet groups if there are no visible facets`, () => {
+      component.searchResult.facets = mockFacets.filter(
+        facet => !facet.visible
+      );
+      fixture.detectChanges();
+
+      const facetGroups = element.queryAll(By.css('.y-search-facet-group'));
+      expect(facetGroups.length).not.toBeGreaterThan(0);
+    });
+
+    it(`should show correct number of facet groups`, () => {
+      component.searchResult.facets = mockFacets.filter(facet => facet.visible);
```

Comment:
Don't filter them. Giving full list will test if only visible facets are shown. With that change, you can also remove the previous test for checking if nothing is shown when facets should not be visible.